tokio-trace/tokio-trace-core/src/dispatcher.rs

```diff
@@ -89,7 +123,15 @@ where
         .try_with(|state| {
             if state.can_enter.replace(false) {
                 let _guard = Entered(&state.can_enter);
-                f(&state.default.borrow())
+
+                let default = &state.default.borrow();
+
+                if default.is::<NoSubscriber>() && GLOBAL_INIT.load(Ordering::SeqCst) == INITIALIZED
```

Comment:
Should we consider setting the thread-local default to the global default in this case? That might improve performance on subsequent calls to `get_default`. Of course, we would want to test this.

Reply:
Oh, good point. Yes, that probably will improve performance, will do.

Reply:
Correctness sketch: because the global default, once set, cannot be changed, all subsequent hits to thread_local NoSubscribers will always fall through to the global default. So, it's fine to cache it locally.

Reply:
We could also check if there's a global default when _exiting_ a scoped default dispatcher, and set the thread-local to that if there is, rather than always setting it back to `NoSubscriber`. I'd expect entering/exiting a dispatcher scope to not be very common, but _getting_ the default is fairly hot.